docs/benchmarkdotnet.md

```diff
@@ -0,0 +1,332 @@
+# BenchmarkDotNet
+
+BenchmarkDotNet is the benchmarking tool that allows to run benchmarks for .NET, .NET Core, CoreRT and Mono. Including private runtime builds.
+
+## Table of Contents
+
+- [Main Concepts](#Main-Concepts)
+- [Running the Benchmarks](#Running-the-Benchmarks)
+  - [Prerequisites](#Prerequisites)
+  - [Interactive Mode](#Interactive-Mode)
+  - [Command Line](#Command-Line)
+    - [Filtering the Benchmarks](#Filtering-the-Benchmarks)
+    - [Listing the Benchmarks](#Listing-the-Benchmarks)
+- [Reading the Results](#Reading-the-Results)
+  - [Reading the Histogram](#Reading-the-Histogram)
+  - [Reading Memory Statistics](#Reading-Memory-Statistics)
+- [Multiple Runtimes](#Multiple-Runtimes)
+- [Regressions](#Regressions)
+- [Profiling](#Profiling)
+- [Disassembly](#Disassembly)
+- [Private Runtime Builds](#Private-Runtime-Builds)
+  - [Running In Process](#Running-In-Process)
+  - [CoreRun](#CoreRun)
+  - [dotnet cli](#dotnet-cli)
+  - [Private CLR Build](#Private-CLR-Build)
+  - [Private CoreRT Build](#Private-CoreRT-Build)
+
+
+## Main Concepts
+
+Benchmarking is really hard (especially microbenchmarking), you can easily make a mistake during performance measurements.
+BenchmarkDotNet will protect you from the common pitfalls (even for experienced developers) because it does all the dirty work for you:
+
+* it generates an isolated project per runtime (with boilerplate code)
+* it builds the project in `Release` (using Roslyn or dotnet cli)
+* it runs every benchmark in a stand-alone process (to achieve process isolation and avoid side effects)
+* it estimates the perfect invocation count per iteration (based on `IterationTime`)
+* it warms-up the code
+* it evaluates the overhead
+* it runs multiple iterations of the method until the requested level of precision is met
+* it consumes the benchmark result to avoid dead code elimination
+* it prevents from inlining of the benchmark by wrapping it with a delegate
+* it prints the results to the console in GitHub markdown, so you can just copy-paste the printed table to GitHub
+* it exports the results to `.\BenchmarkDotNet.Artifacts\results` so you can store them for later use.
+
+A few useful links for you:
+
+* If you want to know more about BenchmarkDotNet features, check out the [Overview Page](http://benchmarkdotnet.org/Overview.htm).
+* If you want to use BenchmarkDotNet for the first time, the [Getting Started](http://benchmarkdotnet.org/GettingStarted.htm) will help you.
+* If you want to ask a quick question or discuss performance topics, use the [gitter](https://gitter.im/dotnet/BenchmarkDotNet) channel.
+
+## Running the Benchmarks
+
+### Prerequisites
+
+To run the benchmarks, you need to [download dotnet cli](./prerequisites.md) first.
```

Comment:
nit: Something like this?
In order to build or run the benchmarks you will need the **.NET Core command-line interface (CLI) tools**. For more information please refer to the [prerequisites](./prerequisites.md).